contracts/Marketplace.sol

```diff
@@ -0,0 +1,164 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.9;
+
+import "./Fisch.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/utils/math/SafeMath.sol";
+
+contract Marketplace is ReentrancyGuard {
+    using SafeMath for uint256;
+
+    Fisch assetNftContract;
+
+    // EVENTS
+    event AuctionCreated(
+        uint256 auctionId,
+        uint256 tokenId,
+        address seller,
+        uint256 startTime,
+        uint256 endTime,
+        uint256 reservePrice,
+        bool started
+    );
+    event PlacedBid(uint256 bidAmount, address bidder, uint256 bidTime, uint256 auctionId, uint256 tokenId);
+    event AuctionCancelled(uint256 auctionId, uint256 tokenId, address seller);
+    event AuctionEnded(uint256 auctionId, address winner, uint256 settledPrice);
+    event AmountSent(address indexed to, uint256 indexed amount);
+    event AmountReceived(address sender, uint256 amount);
+
+    uint256 minimumBid = 1e18;
+
+    struct Auction {
+        uint256 tokenId;
+        address seller;
+        uint256 startTime;
+        uint256 endTime;
+        uint256 reservePrice;
+        bool started;
+    }
+
+    mapping(uint256 => Auction) public auctions;
+    uint256 public nextAuctionId = 0;
+
+    struct HighestBidder {
+        address bidder;
+        uint256 bid;
+        uint256 bidTime;
+    }
+    mapping(uint256 => HighestBidder) public highestBids;
+
+
+    constructor(address _assetNft) {
+        assetNftContract = Fisch(_assetNft);
+    }
+
+    function fetchAuction() public {}
```

Comment:
this would be just to fetch the auction struct by it's Id

Reply:
ok, I get it!